docs/registries/IndexRegistry.md

```diff
@@ -4,85 +4,163 @@
 | -------- | -------- | -------- |
 | [`IndexRegistry.sol`](../../src/IndexRegistry.sol) | Singleton | Transparent proxy |
 
-TODO
+The `IndexRegistry` provides an index for every registered Operator in every quorum. For example, if a quorum has `n` Operators, every Operator registered for that quorum will have an index in the range `[0:n-1]`. The role of this contract is to provide an AVS with a common, on-chain ordering of Operators within a quorum.
 
-#### High-level Concepts
+*In EigenDA*, the Operator ordering properties of the `IndexRegistry` will eventually be used in proofs of custody, though this feature is not implemented yet.
 
-TODO
+#### Important State Variables
 
-This document organizes methods according to the following themes (click each to be taken to the relevant section):
+```solidity
+/// @notice maps quorumNumber => operator id => current index
+mapping(uint8 => mapping(bytes32 => uint32)) public currentOperatorIndex;
+
+/// @notice maps quorumNumber => index => historical operator ids at that index
+mapping(uint8 => mapping(uint32 => OperatorUpdate[])) internal _indexHistory;
+
+/// @notice maps quorumNumber => historical number of unique registered operators
+mapping(uint8 => QuorumUpdate[]) internal _operatorCountHistory;
+ 
+struct OperatorUpdate {
+    uint32 fromBlockNumber;
+    bytes32 operatorId;
+}
+
+struct QuorumUpdate {
+    uint32 fromBlockNumber;
+    uint32 numOperators;
```

Comment:
Yeah, agreed, it's been bugging me a bit. The other registries call this field `updateBlockNumber` - would be good to at least make this consistent with those.

But since that's a code change, I'm not gonna include it in this PR.